File: trackit_app/lib/utils/requirement_file_types.dart
/// Shared between student requirement submissions and instructor
/// template uploads -- both are restricted to real office-document
/// formats, matching the server's requirementFileFilter.
const List<String> requirementFileExtensions = ['pdf', 'doc', 'docx', 'xls', 'xlsx'];

String requirementFileContentType(String? extension) {
  switch (extension?.toLowerCase()) {
    case 'pdf':
      return 'application/pdf';
    case 'doc':
      return 'application/msword';
    case 'docx':
      return 'application/vnd.openxmlformats-officedocument.wordprocessingml.document';
    case 'xls':
      return 'application/vnd.ms-excel';
    case 'xlsx':
      return 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet';
    default:
      return 'application/octet-stream';
  }
}

File: trackit_app/server/utils/requirementFileTypes.js
const ALLOWED_MIME_TYPES = new Set([
  'application/pdf',
  'application/msword',
  'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
  'application/vnd.ms-excel',
  'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
]);

// Requirement documents (both student submissions and instructor
// templates) are restricted to real office-document formats -- unlike
// e.g. announcement attachments, there's no reason a requirement file
// would ever be an image.
function requirementFileFilter(req, file, cb) {
  cb(null, ALLOWED_MIME_TYPES.has(file.mimetype));
}

module.exports = { requirementFileFilter };
